sdb/commands/linux/ps.py

```diff
@@ -0,0 +1,180 @@
+                datetime.timedelta(seconds=int(obj.stime) / 1000 / 1000)),
+        "PPID":
+            lambda obj: int(obj.parent.pid),
+        "STAT":
+            lambda obj: str(Stacks.task_struct_get_state(obj)),
+        "CMD":
+            lambda obj: str(obj.comm.string_().decode("utf-8")),
+    }
+
+    @classmethod
+    def _init_parser(cls, name: str) -> argparse.ArgumentParser:
+        parser = super()._init_parser(name)
+        parser.add_argument('-e', '--every', action='store_true', \
+                help="Select all processes. Identical to -A")
+        parser.add_argument('-A', '--all', action='store_true', \
+                help="Select all processes. Identical to -e")
+        parser.add_argument('-C', '--C', type=str, \
+                help="Print only the process IDs of a command")
+        parser.add_argument('-x', '--x', action='store_true', \
+                help="Show PID, TIME, CMD")
+        parser.add_argument('-h', action='store_true', \
+                help="Show the output without headers.")
+        parser.add_argument('-p', '--p', type=str, \
+                help="Select by process ID. Identical to --pid.")
+        parser.add_argument('-ppid', '--ppid', type=str, \
+                help="Select by parent process ID.  \
+                      This selects the processes with \
+                      a parent process ID in the pid list."
+                    )
+        parser.add_argument('-o',
+                            '--format',
+                            type=str,
+                            help="User-defined format. \
+                            format is a single argument in the form of a blank-separated \
+                            or comma-separated list, which offers a way to specify individual\
+                            output columns. Headers may be renamed (ps -o pid,ppid) as desired.  "
+                           )
+        return parser
+
+    def pretty_print(self, objs: Iterable[drgn.Object]) -> None:
+        fields = list(Ps.FIELDS.keys())
+
+        fields_option_ae = [
+            x for x in fields if x in "UID, PID, PPID, STIME, TIME, CMD"
+        ]
+        fields_option_x = [x for x in fields if x in "PID, STAT, TIME, CMD"]
+        fields_option_default = [x for x in fields if x in "PID, TIME, CMD"]
+
+        if self.args.every or self.args.all:
+            fields = fields_option_ae
+        elif self.args.x:
+            fields = fields_option_x
+        elif self.args.format:
+            fields_option_o = 'TASK,' + self.args.format.upper()
+            fields_option_o = [x for x in fields if x in fields_option_o]
+            fields = fields_option_o
+        else:
+            fields = fields_option_default
+
+        table = Table(fields, None, {"PID": str})
+
+        cmds = self.args.C.split(',') if self.args.C else []
+        pids = [int(x) for x in self.args.p.split(',')] if self.args.p else []
+        ppids = [int(x) for x in self.args.ppid.split(',')
```

Comment:
what happens here when we get faulty input? (e.g. "asdf, asdfasdf")

If the result is that we blow up maybe it may be easier to switch from the `ps --pid 1,2,3` to `ps --pid 1 2 3` notation (both are supported from the userland `ps` utility in Linux from what I can tell). This would mean that we'd need to change the argument type to something like `type=int, nargs="+",`.

Reply:
Nice catch - I'll update.